fluss-server/src/main/java/org/apache/fluss/server/coordinator/CoordinatorService.java

```diff
@@ -366,8 +373,36 @@ private void handleFlussTableChanges(
             }
         }
 
+        TableInfo existTableInfo = metadataManager.getTable(tablePath);
+        TableDescriptor existTableDescriptor = existTableInfo.toTableDescriptor();
+
         metadataManager.alterTableProperties(
                 tablePath, propertyChangesBuilder.build(), ignoreIfNotExists);
+
+        TableInfo alteredTableInfo = metadataManager.getTable(tablePath);
+        TableDescriptor alteredTableDescriptor = alteredTableInfo.toTableDescriptor();
+        // enable lake table
+        if (!isDataLakeEnabled(existTableDescriptor) && isDataLakeEnabled(alteredTableDescriptor)) {
+            // TODO: should tolerate if the lake exist but matches our schema. This ensures
+            // eventually
+            //  consistent by idempotently creating the table multiple times. See #846
+            // before create table in fluss, we may create in lake
+            try {
+                checkNotNull(lakeCatalog).createTable(tablePath, alteredTableDescriptor);
```

Comment:
actaully, I'd suggest first create the lake table and then alter table properites. 
The reason is that `create the lake table` is more like to fail, and if we alreay alter table properties, and then create lake table fail. It will then case although the table is datalake enabeld, but the lake table is not created.